backend/src/main/kotlin/org/icpclive/service/QueueService.kt

```diff
@@ -43,14 +46,23 @@ class QueueService {
         })
     }
 
+    private suspend fun modifyRun(run: RunInfo) {
+        resultFlow.emit(if (run.id in runs) ModifyRunInQueueEvent(run) else AddRunToQueueEvent(run))
+        runs[run.id] = run
+    }
+
     private suspend fun removeRun(run: RunInfo) {
         runs.remove(run.id)
+        removedRuns[run.id] = run
         resultFlow.emit(RemoveRunFromQueueEvent(run))
     }
 
     private val RunInfo.timeInQueue
-        get() = if (isFirstSolvedRun) FIRST_TO_SOLVE_WAIT_TIME else WAIT_TIME
-
+        get() = when {
+            isFeaturedRun -> FEATURED_WAIT_TIME
```

Comment:
And what if featured run is also first to solve?

Reply:
have i to sum times?

Reply:
No idea, but current behaviour is strange. Many
Maybe @lperovskaya knows how it should work.